src/integer_mod_q/modulus_polynomial_ring_zq/get.rs

```diff
@@ -8,11 +8,124 @@
+    /// let coeff_1: Zq = poly.get_coeff(1).unwrap();
+    /// let coeff_4: Zq = poly.get_coeff(4).unwrap();
+    ///
+    /// assert_eq!(Zq::from((0, 17)), coeff_0);
+    /// assert_eq!(Zq::from((1, 17)), coeff_1);
+    /// assert_eq!(Zq::from((0, 17)), coeff_4);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds) if
+    ///     either the index is negative or it does not fit into an [`i64`].
+    fn get_coeff(&self, index: impl TryInto<i64> + Display) -> Result<Zq, MathError> {
+        let index = evaluate_index(index)?;
+        let mut out_z = Z::default();
+        let mut ctx = MaybeUninit::uninit();
+
+        unsafe {
+            fmpz_mod_poly_get_coeff_fmpz(
+                &mut out_z.value,
+                &self.modulus.modulus[0],
+                index,
+                &self.get_fq_ctx_struct().ctxp[0],
+            );
+
+            fmpz_mod_ctx_init(ctx.as_mut_ptr(), &self.get_fq_ctx_struct().ctxp[0].n[0]);
+
+            let modulus = Modulus {
+                modulus: Rc::new(ctx.assume_init()),
+            };
+
+            Ok(Zq::from((out_z, modulus)))
+        }
+    }
+}
+
+impl GetCoefficient<Z> for ModulusPolynomialRingZq {
+    /// Returns the coefficient of a polynomial [`ModulusPolynomialRingZq`] as a [`Z`].
+    ///
+    /// If an index is provided which exceeds the highest set coefficient, `0` is returned.
+    ///
+    /// Parameters:
+    /// - `index`: the index of the coefficient to get (has to be positive)
+    ///
+    /// Returns the coefficient as a [`Z`], or a [`MathError`] if the provided index
+    /// is negative and therefore invalid, or it does not fit into an [`i64`].
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::traits::*;
+    /// use qfall_math::integer_mod_q::ModulusPolynomialRingZq;
+    /// use qfall_math::integer::Z;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = ModulusPolynomialRingZq::from_str("4  0 1 2 3 mod 17").unwrap();
+    ///
+    /// let coeff_0: Z = poly.get_coeff(0).unwrap();
+    /// let coeff_1: Z = poly.get_coeff(1).unwrap();
+    /// let coeff_4: Z = poly.get_coeff(4).unwrap();
+    ///
+    /// assert_eq!(Z::ZERO, coeff_0);
+    /// assert_eq!(Z::ONE, coeff_1);
+    /// assert_eq!(Z::ZERO, coeff_4);
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`OutOfBounds`](MathError::OutOfBounds) if
+    ///     either the index is negative or it does not fit into an [`i64`].
+    fn get_coeff(&self, index: impl TryInto<i64> + Display) -> Result<Z, MathError> {
+        let index = evaluate_index(index)?;
+        let mut out = Z::default();
+
+        unsafe {
+            fmpz_mod_poly_get_coeff_fmpz(
+                &mut out.value,
+                &self.modulus.modulus[0],
+                index,
+                &self.get_fq_ctx_struct().ctxp[0],
+            )
+        }
+
+        Ok(out)
```

Comment:
wet code -> you have written the same code twice. I would suggest to find a way to reduce the number of lines, e.g. by calling these lines in the other function or vice-versa